barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.hpp

```diff
@@ -251,6 +261,16 @@ class AvmTraceBuilder {
     uint32_t side_effect_counter = 0;
     uint32_t external_call_counter = 0; // Incremented both by OpCode::CALL and OpCode::STATICCALL
     ExecutionHints execution_hints;
+    // These are the tracked roots for intermediate steps
+    TreeSnapshots intermediate_tree_snapshots;
+    // These are some counters for the tree acceess hints that we probably dont need in the future
+    uint32_t note_hash_read_counter = 0;
+    uint32_t note_hash_write_counter = 0;
+    uint32_t nullifier_read_counter = 0;
+    uint32_t nullifier_write_counter = 0;
+    uint32_t l1_to_l2_msg_read_counter = 0;
+    uint32_t storage_read_counter = 0;
+    uint32_t storage_write_counter = 0;
 
```

Comment:
We actually _might_ need these in the future because we need to raise error flag if these surpass some limits.